file/src/find.rs

```diff
@@ -0,0 +1,400 @@
+                }
+                Expr::Size(size, in_bytes) => {
+                    if let Ok(metadata) = file.metadata() {
+                        let file_size = if *in_bytes {
+                            metadata.len()
+                        } else {
+                            (metadata.len() + 511) / 512
+                        };
+                        if file_size < *size {
+                            c_files.remove(file.path());
+                        }
+                    }
+                }
+                Expr::Newer(f) => {
+                    if let Ok(metadata) = fs::metadata(f) {
+                        if let Ok(file_metadata) = file.metadata() {
+                            if !(file_metadata.modified().unwrap() > metadata.modified().unwrap()) {
+                                c_files.remove(file.path());
+                            }
+                        }
+                    }
+                }
+                Expr::Print if c_files.contains(file.path()) => {
+                    result.push(file.path().to_path_buf());
+                }
+                Expr::Print if !c_files.contains(file.path()) => {
+                    continue;
+                }
+                _ => return Err("Error: Invalid expression".to_string()),
+            }
+        }
+    }
+
+    if result.is_empty() {
+        result.extend(c_files.clone());
+    }
+    result.sort();
+    Ok(result)
+}
+
+/// Retrieves the root path from a list of expressions.
+///
+/// # Arguments
+///
+/// * `expr` - A slice of `Expr` expressions.
+///
+/// # Returns
+///
+/// * A `String` representing the root path extracted from the expressions.
+fn get_root(expr: &[Expr]) -> String {
+    let mut path = String::new();
+    for i in expr {
+        match i {
+            Expr::Path(p) => path = p.to_string(),
+            _ => continue
+        } 
+    }
+    path
+}
+
+/// Executes the find command with the provided arguments.
+///
+/// # Arguments
+///
+/// * `args` - A vector of `String` representing the command-line arguments.
+///
+/// # Returns
+///
+/// * A `Result` indicating success or containing an error message as a `String`.
+fn find(args: Vec<String>) -> Result<(), String> {
+    let mut tokens: Vec<&str> = args.iter().skip(1).rev().map(|s| s.as_str()).collect();
+    let binding = parse_expression(&mut tokens);
+    let expr = binding.as_slice();
+    let path = get_root(expr);
+
+    let root_dev = if let Ok(metadata) = fs::metadata(path.clone()) {
+        metadata.dev()
+    } else {
+        return Err("Error: Could not retrieve root device metadata".to_string());
+    };
+
+    let files = WalkDir::new(path).into_iter().map(|f|f.unwrap()).collect::<Vec<DirEntry>>();
+    let result = evaluate_expression(expr, files, root_dev);
+
+    if result.is_ok() {
+        for res in result.unwrap() {
+            println!("{}", res.display())
+        }
+        Ok(())
+    }
+    else {
+        Err(result.err().unwrap())
+    }
```

Comment:
Just propagate the error here.
```suggestion
        for res in result? {
            println!("{}", res.display())
        }
        Ok(())
```